### ISSUE TITLE
Rubberduck.1.22 64 bit installed but not visible unless run as local admin

### ISSUE THREAD

**Issue by emailgregn** [rubberduck-vba/Rubberduck#629]
I had to get a sysadmin to run the installer with his credentials because our machines are locked down.
The addin then doesn't appear for my regular user. When running Excel from an elevated local admin account, Rubberduck works fine.

**Comment by ckuhn203** [comment]
Ahhh now I see. It installed the registry keys under the admin's HKCU. They need to be installed under yours. Will update as soon as possible. I read an MZTools article about this once upon a time. I need to find it again.

**Comment by ckuhn203** [comment]
After reading this, I'm pretty sure that you'll need to use your temporarily elevated admin access to run the install. 

http://www.mztools.com/v3/faq.aspx

Note that I don't believe the second workaround will work. I don't think we've set things up to allow a local machine install in that version. We'll double check on this prior to our next release.

**Comment by ckuhn203** [comment]
Related #458 & #589

**Comment by rossknudsen** [comment]
> ...you'll need to use your temporarily elevated admin access...
You need to temporarily make your normal user account an administrator.  Most people will find this is not feasible.

I was actually going to link the MZ-Tools FAQ here but you beat me to it.  It would seem it does a similar thing to the current RD installer.  Registers the classes for COM interop using regasm **for all users**.  Then adds the Office addin keys to the **installing user**'s HKCU hive.  So when you elevate your privileges from your normal account to an admin account, you change the installing user to the admin user.  I know I'm stating the obvious.

Installing for all users
---------------------------
If we want to install the addin for all users, then the current COM registration is what we want.  We just need to change the Office addin registration.  As far as I have been able to determine, Office does not check HKLM.  The following quote is from [MZ-Tools blog](http://www.mztools.com/articles/2012/MZ2012018.aspx).

> ... the HKEY_LOCAL_MACHINE hive is ignored by the VBA editor when looking for add-ins ...

I've also tried to do this by adding keys to the registry by hand and have not been successful either.  If anyone finds a way to make this work for all users, then let me know because this option would probably be the best one.

Installing for just the current user
------------------------------------------
If we want to install the addin for just the current user, then we need to make a couple of changes.  Firstly, we would like to change the permissions needed to run the installer.  If we could run as the local non-admin user, then we wouldn't have a problem with the Office addin keys being registered under the wrong user's HKCU hive.  So to achieve this we would need to be able to perform the COM registration without elevated permissions.  The following links are some possible approaches to achieve this:

* [regasm with the /regfile option, and replacing HKLM with HKCU in the resulting .reg file](http://stackoverflow.com/a/1613018/2301065)
* [One of the solutions here](http://stackoverflow.com/q/23063552/2301065)

Basically we end up creating keys in the HKCU hive instead of HKLM since the latter requires admin rights.  This is completely untested, but will probably require taking more direct charge over the COM registration than leaving it up to regasm.

**Comment by rossknudsen** [comment]
Referencing #589 

> Could it be because of the Flags: runascurrentuser flag?

I just [RTM](http://www.jrsoftware.org/ishelp/index.php?topic=runsection) and there is a runasoriginaluser flag which can be used instead.  This is only applicable to items in the Run section which means it needs to be associated with something you would run from the command line.

**Comment by retailcoder** [comment]
@ckuhn203 @rossknudsen this was fixed, right? I mean, we have a work-around?

**Comment by rossknudsen** [comment]
Yeah, there is a work-around by creating the reg keys manually under the correct hive.

A proper fix would be to create the COM registration keys under HKCU eliminating the need for admin privileges completely.  This requires manual registration rather than using regasm though.

**Comment by retailcoder** [comment]
Manual registration is out of the question, at least as far as I'm concerned.
Rubberduck 1.4.3 has been downloaded 785 times as of this writing, and only a handful of those had issues running the add-in. The PowerShell script I added to the release notes is good enough for MZ-Tools and that's good enough for me (see #800).


Sent from my Samsung device

**Comment by manjugb** [comment]
Hi , 
Is this possible to add Rubber into visual studio community 2017, I installed the rubberduck.exe file on my window , how can i access or use in visual studio 2017

**Comment by bclothier** [comment]
@manjugb It's not clear what you intend to do here.

If you are trying to use Rubberduck, it's a VBE add-in, not an add-in for Visual Studio, so you should use a VBA host (e.g. Excel, Access, or anything else that let you writes VBA code)

If you are trying to contribute to Rubberduck, then you need to fork & clone the Rubberduck git repository so that you can then load the project into the Visual Studio.

**Comment by retailcoder** [comment]
Hi @manjugb,

If you mean to *use* Rubberduck, then you're all set (having installed a release build with admin privs) - just start Excel, bring up the VBE and Rubberduck will load.

If you mean to help *develop* Rubberduck, then you need to uninstall the release build and fork/clone the repository through git. See the [wiki](https://github.com/rubberduck-vba/Rubberduck/wiki/Contributing) and read [CONTRIBUTING.MD](https://github.com/rubberduck-vba/Rubberduck/blob/next/CONTRIBUTING.md) - don't hesitate to ask if you have any questions!
